TreasureData/TreasureData.m

```diff
@@ -694,6 +697,68 @@ - (void)resetUniqId {
 }
 
+#pragma mark - Personalization API
+
+- (void)fetchUserSegments: (nonnull NSArray<NSString *> *)audienceTokens
+                     keys: (nonnull NSDictionary<NSString *, id> *)keys
+                  options: (nullable NSDictionary<TDRequestOptionsKey, id> *)options
+        completionHandler: (void (^_Nonnull)(NSArray* _Nullable jsonResponse, NSError* _Nullable error)) handler {
+    NSString *cdpEndpoint = self.cdpEndpoint ? self.cdpEndpoint : defaultCdpEndpoint;
+    NSString *audienceString = [NSString
+                                stringWithFormat:@"&token=%@",
+                                [audienceTokens componentsJoinedByString: @","]];
+    NSMutableString *keyString = [[NSMutableString alloc] initWithString: @""];
+    for (NSString *key in keys) {
+        [keyString appendFormat:@"&key.%@=%@", key, keys[key]];
+    }
+    NSMutableString *urlString = [NSMutableString stringWithString:cdpEndpoint];
+    [urlString appendString:@"/cdp/lookup/collect/segments?version=2"];
+    [urlString appendString:audienceString];
+    [urlString appendString:keyString];
+    
+    NSNumber *timeoutNumber = (NSNumber *)options[TDRequestOptionsTimeoutIntervalKey];
+    NSTimeInterval timeout = timeoutNumber ? [timeoutNumber doubleValue] : 60;
+    NSNumber *cachePolicyNumber = (NSNumber *)options[TDRequestOptionsCachePolicyKey];
+    NSURLRequestCachePolicy cachePolicy = cachePolicyNumber ? [cachePolicyNumber unsignedIntegerValue] : NSURLRequestUseProtocolCachePolicy;
+    NSURL *url = [NSURL URLWithString:urlString];
+    NSURLRequest *urlRequest = [NSURLRequest requestWithURL:url
+                                                cachePolicy:cachePolicy
+                                            timeoutInterval:timeout];
+    NSOperationQueue *queue = [NSOperationQueue currentQueue];
+    [NSURLConnection sendAsynchronousRequest:urlRequest queue:queue completionHandler:^(NSURLResponse * _Nullable response, NSData * _Nullable data, NSError * _Nullable connectionError) {
+        if (connectionError) {
+            handler(nil, connectionError);
+        } else {
+            NSError *jsonError = nil;
+            id jsonResponse = [NSJSONSerialization JSONObjectWithData:data
+                                                              options:kNilOptions
+                                                                error:&jsonError];
+            if ([jsonResponse isKindOfClass: [NSArray class]]) {
+                // Successful case
+                handler((NSArray *)jsonResponse, jsonError);
+            } else if ([jsonResponse isKindOfClass: [NSDictionary class]] && ((NSDictionary *)jsonResponse)[@"error"] != nil) {
+                // Error returned in response
+                NSDictionary *errorResponse = (NSDictionary *)jsonResponse;
+                NSDictionary *userInfo = @{
+                   NSLocalizedDescriptionKey: NSLocalizedString(errorResponse[@"error"], nil),
```

Comment:
Just my 2 cents: consider making the error simply `NSString` instead. Localized version might make developers falsely assume these are actually localized while they're not.

Reply:
Aggree